rules/explicit_module/sdk_clang_module.bzl

```diff
@@ -1,47 +1,103 @@
+    "SwiftInfo",
+    "SwiftToolchainInfo",
+    "create_clang_module",
+    "create_module",
+    "create_swift_info",
+)
 load("@build_bazel_rules_swift//swift/internal:swift_common.bzl", "swift_common")
+load(
+    "@build_bazel_rules_swift//swift/internal:utils.bzl",
+    "compilation_context_for_explicit_module_compilation",
+    "get_providers",
+)
 
 def _sdk_clang_module_impl(ctx):
+    compilation_context = cc_common.create_compilation_context()
+    compilation_context_to_compile = (
+        compilation_context_for_explicit_module_compilation(
+            compilation_contexts = (
+                [compilation_context]
+            ),
+            deps = ctx.attr.deps,
+        )
+    )
+
+    swift_toolchain = ctx.attr._toolchain[SwiftToolchainInfo]
+    requested_features = ctx.features + [SWIFT_FEATURE_SYSTEM_MODULE]
+    feature_configuration = swift_common.configure_features(
+        ctx = ctx,
+        requested_features = requested_features,
+        swift_toolchain = swift_toolchain,
+        unsupported_features = ctx.disabled_features,
+    )
+
+    swift_infos = get_providers(ctx.attr.deps, SwiftInfo)
+
+    precompiled_module = precompile_clang_module(
+        actions = ctx.actions,
+        cc_compilation_context = compilation_context_to_compile,
+        feature_configuration = feature_configuration,
+        module_map_file = ctx.attr.module_map,
+        module_name = ctx.attr.module_name,
+        swift_infos = swift_infos,
+        swift_toolchain = swift_toolchain,
+        target_name = ctx.attr.name,
+    )
+
     return [
-        swift_common.create_swift_interop_info(
-            module_map = ctx.attr.module_map,
-            module_name = ctx.attr.module_name,
-            requested_features = [SWIFT_FEATURE_SYSTEM_MODULE],
+        create_swift_info(
+            modules = [
+                create_module(
+                    name = ctx.attr.module_name,
+                    clang = create_clang_module(
+                        compilation_context = compilation_context,
+                        module_map = ctx.attr.module_map,
+                        precompiled_module = precompiled_module,
+                    ),
+                ),
+            ],
+            swift_infos = swift_infos,
+        ),
+        OutputGroupInfo(
+            swift_explicit_module = depset([precompiled_module]),
         ),
-        # We need to return CcInfo and its compilation_context. We may also consider to update swift_clang_module_aspect.
-        # See https://github.com/bazelbuild/rules_swift/blob/d68b21471e4e9d922b75e2b0621082b8ce017d11/swift/internal/swift_clang_module_aspect.bzl#L548
-        CcInfo(compilation_context = cc_common.create_compilation_context()),
-        # Required to add sdk_clang_module targets to the deps of swift_module_alias.
-        # TODO(cshi): create the SwiftInfo correctly
-        create_swift_info(),
     ]
 
 sdk_clang_module = rule(
```

Comment:
i notice that the refence to sdk_clang_module is removed in the example... that is because those are code gen by the repo rules right?